gravitee-policy-ratelimit/src/main/java/io/gravitee/policy/ratelimit/RateLimitPolicy.java

```diff
@@ -151,17 +156,29 @@ private static Completable errorManagement(
         if (throwable instanceof PolicyRateLimitException ex) {
             return Completable.error(ex);
         }
-        // Set Rate Limit headers on response
-        if (rateConfig.isAddHeaders()) {
-            ctx.response().headers().set(X_RATE_LIMIT_LIMIT, Long.toString(limit));
-            // We don't know about the remaining calls, let's assume it is the same as the limit
-            ctx.response().headers().set(X_RATE_LIMIT_REMAINING, Long.toString(limit));
-            ctx.response().headers().set(X_RATE_LIMIT_RESET, Long.toString(-1));
-        }
-        ctx.warnWith(
-            new ExecutionWarn(RATE_LIMIT_NOT_APPLIED).message("Request bypassed rate limit policy due to internal error").cause(throwable)
-        );
-        // If an errors occurs at the repository level, we accept the call
-        return Completable.complete();
+        return switch (getRateLimitPolicyConfiguration().getStrictnessStrategy().strictness()) {
+            case STRICT -> Completable.error(
+                PolicyRateLimitException.overflow(
+                    RateLimitPolicyV3.RATE_LIMIT_TOO_MANY_REQUESTS,
+                    "Request blocked by rate limit policy due to internal error",
+                    Map.of()
+                )
+            );
```

Comment:
I think we should use `ctx.interruptWith(...)` with a proper ExecutionFailure that holds a custom exception with the message "Request blocked by rate limit policy due to internal error". This message will be visible by the API Publisher only. The failure.message is for the end user and should not indicate any issue.